tests/virtual_machines_test.go

```diff
@@ -517,42 +517,6 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			})
 		})
 
-		Context("When the leader is changed", func() {
```

Comment:
Maybe it's  not directly relevant to this PR, but we should add another test, to check that kmp internal cache (of allocated macs) "migrates" safely (it doesn't really migrate but re-constructed, but still this is something we should make sure that happens).